### PR TITLE
Add documentation for profile selectors

### PR DESCRIPTION
**Pull request by eleftherias**
# Summary

Ref #4018

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [x] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: docs/docs/how-to/profile_selectors.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  - entity: repository
+    selector: repository.is_fork != true && repository.name.startsWith('stacklok/')
+  - entity: artifact
+    selector: artifact.provider.name == 'github-app-stacklok'
+```
+
+Let's break down the example above:
+- `entity`: Defines the type of entity you want to filter (`repository`, `artifact`, or `pull_request`).
+- `selector`: The CEL expression that specifies the filtering criteria. In the example:
+  - The first selector filters repositories to include only those that are not forks and whose name starts with `stacklok/`, in other words those that are part of the stacklok organization.
```
Comment (mesembria) [inline docs/docs/how-to/profile_selectors.md:38]: suggestion : The first selector filters repositories to include only those that are not forks and whose name starts with `stacklok/`. In other words, those that are part of the stacklok organization.

File: docs/docs/how-to/profile_selectors.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Below you can find the full list of selectors available for each entity type.
+
+## Repository selectors
+
+Selectors for repositories allow you to filter and manage repositories based on specific attributes.
+
+| Field        | Description                                                                                         | Type             |
+|--------------|-----------------------------------------------------------------------------------------------------|------------------|
+| `name`       | The full name of the repository, e.g. stacklok/minder                                               | string           |
```
Comment (mesembria) [inline docs/docs/how-to/profile_selectors.md:49]: question: Do we want to explicitly call out that wildcards and partials are not supported, or do you feel this is sufficient?

File: docs/docs/how-to/profile_selectors.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+| `is_fork`    | `true` if the repository is a fork, `nil` if unknown or not applicable to this provider             | bool             |
+| `is_private` | `true` if the repository is private, `nil` if unknown or not applicable to this provider            | bool             |
+| `provider`   | The provider of the repository, for me details see [Provider Selectors](#entity-provider-selectors) | ProviderSelector |
```
Comment (mesembria) [inline docs/docs/how-to/profile_selectors.md:52]: The provider of the repository, for **more** details see

File: docs/docs/how-to/profile_selectors.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+## Artifact selectors
+
+| Field      | Description                                                                                       | Type             |
+|------------|---------------------------------------------------------------------------------------------------|------------------|
+| `name`     | The full name of the artifact, e.g. stacklok/minder-server                                        | string           |
+| `type`     | The type of the artifact, e.g. "container"                                                        | string           |
+| `provider` | The provider of the artifact, for me details see [Provider Selectors](#entity-provider-selectors) | ProviderSelector |
```
Comment (mesembria) [inline docs/docs/how-to/profile_selectors.md:60]: The provider of the artifact, for **more** details see

File: docs/docs/how-to/profile_selectors.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+|--------|-------------------------------------------------------------|--------|
+| `name` | The full name of the pull request, e.g. stacklok/minder/123 | string |
+
+## Entity provider selectors
+
+Each entity can be filtered based on its provider.
+
+| Field   | Description                                        | Type   |
+|---------|----------------------------------------------------|--------|
+| `name`  | the name of the provider, e.g. github-app-stacklok | string |
```
Comment (mesembria) [inline docs/docs/how-to/profile_selectors.md:74]: nitpick: Capitalize the first word of the description to be consistent with the rest of the table. 
**The** name of the provider,

File: docs/docs/how-to/profile_selectors.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+name: profile-with-selectors
+selection:
+  - entity: repository
+    selector: repository.is_fork != true && repository.name.startsWith('stacklok/')
+  - entity: artifact
+    selector: artifact.provider.name == 'github-app-stacklok'
+```
+
+Let's break down the example above:
+- `entity`: Defines the type of entity you want to filter (`repository`, `artifact`, or `pull_request`).
```
Comment (jhrozek) [inline docs/docs/how-to/profile_selectors.md:36]: I think this was not written explicitly anywhere in the design or maybe wasn't prominent, but you can also omit the `entity` attribute and just add the `selector`. Then the selector would apply to any entity type with the caveat that the available attributes is at the moment just the name. In the near future we hope to add also 'properties' as a generic key-value map.

**Review by jhrozek — DISMISSED** [review]

File: docs/docs/how-to/profile_selectors.md (hunk 6 above)
Comment (eleftherias) [reply docs/docs/how-to/profile_selectors.md]: Updated the docs to call that out

File: docs/docs/how-to/profile_selectors.md (hunk 2 above)
Comment (eleftherias) [reply docs/docs/how-to/profile_selectors.md]: Any kind of wildcard support would need to be part of CEL, so I've left it out of our docs

**Review by jhrozek — APPROVED** [review]

**Pull request merged**